### PR TITLE
docs: Instructions to make MAVA editable

### LINKED ISSUE #752: [MAINTAIN] Make MAVA repo editable
### Please describe what needs to be maintained?
When a user installs the mava directory and runs an example, Python frequently calls mava from the site-package rather than the mava directory that was cloned : When the user runs an example after making modifications to files in the MAVA directory, Python will call MAVA from the site-packages (Call the package as if the changes on the library never happened)

### Describe the outcome you'd like
Update documentation pages with the required instructions to avoid this happening.

### PR DESCRIPTION
**Pull request by OmaymaMahjoub**
## What?
Write docs of common errors that occur when installing Mava (like no module named mava, running mava as a package and not directory...).

## Extra
Close #752

### CODE REVIEW

File: docs/gotchas.md [hunk 1]
```diff
@@ -40,3 +40,8 @@ class DistributorConfig:
 class Distributor(Component):
     def __init__(self, config: DistributorConfig = DistributorConfig()):
 ```
+
+## Installing from source
+- In order to edit the MAVA source code, you must ensure that you have cloned the repo and installed from source using `pip install -e .`
+- A common mistake is to `run pip install id-mava`, which installs MAVA from pypi, resulting in a duplicate version of MAVA being stored as a dependancy.
+- If you encounter `ModuleNotFoundError: No module named 'mava'` or Mava is executed by the example systems via site-packages (mava is stored as a package), a likely fix is to add the home directroy of the cloned Mava source code to `PYTHONPATH` with `export PYTHONPATH=<mava_homedir>`. On a linux-based, this command can be added to `.bashrc`, to avoid having to export the path every time a new terminal is opened.
```
Comment (AsadJeewa) [inline docs/gotchas.md:47]: typo: "home DIRECTORY"

Comment (AsadJeewa) [reply docs/gotchas.md:47]: typo: "on a linus-based SYSTEM"

Comment (AsadJeewa) [reply docs/gotchas.md:47]: Think you can remove "Mava is executed by the example systems via site-packages (mava is stored as a package)"  as this is not fixed by PYTHONPATH but the issue is duplicate installation (covered by the bullet point above). Also, if you mention site-packages, you would need to mention venv.

**Review by AsadJeewa — CHANGES_REQUESTED** [review]
Thanks @OmaymaMahjoub 
Small changes requested

File: docs/gotchas.md (hunk 1 above)
Comment (AsadJeewa) [reply docs/gotchas.md:47]: Perhaps you can move it to the bullet point above

Comment (AsadJeewa) [reply docs/gotchas.md]: @omayma typo: "home DIRECTORY" still there

**Review by AsadJeewa — DISMISSED** [review]
Thanks @OmaymaMahjoub

File: docs/gotchas.md [hunk 2]
```diff
@@ -40,3 +40,8 @@ class DistributorConfig:
 class Distributor(Component):
     def __init__(self, config: DistributorConfig = DistributorConfig()):
 ```
+
+## Installing from source
+- In order to edit the MAVA source code, you must ensure that you have cloned the repo and installed from source using `pip install -e .`
```
Comment (arnupretorius) [inline docs/gotchas.md:45]: Minor: "Mava" instead of "MAVA"

File: docs/gotchas.md [hunk 3]
```diff
@@ -40,3 +40,8 @@ class DistributorConfig:
...
+- A common mistake is to `run pip install id-mava`, which installs MAVA from pypi, resulting in a duplicate version of MAVA being stored as a dependancy.
```
Comment (arnupretorius) [inline docs/gotchas.md:46]: Minor: "pypi" -> "PyPI"

**Review by arnupretorius — COMMENTED** [review]
Thanks @OmaymaMahjoub 🙂  See my very minor comments.

File: docs/gotchas.md (hunk 2 above)
Comment (OmaymaMahjoub) [reply docs/gotchas.md]: Done!

File: docs/gotchas.md (hunk 3 above)
Comment (OmaymaMahjoub) [reply docs/gotchas.md]: Done!

File: docs/gotchas.md (hunk 1 above)
Comment (OmaymaMahjoub) [reply docs/gotchas.md]: Done!

**Review by AsadJeewa — DISMISSED** [review]
Thanks @OmaymaMahjoub

File: docs/gotchas.md [hunk 4]
```diff
@@ -40,3 +40,8 @@ class DistributorConfig:
 class Distributor(Component):
     def __init__(self, config: DistributorConfig = DistributorConfig()):
 ```
+
+## Installing from source
+- In order to edit the Mava source code, you must ensure that you have cloned the repo and installed from source using `pip install -e .`
+- A common mistake is to `run pip install id-mava`, which installs MAVA from PyPi, resulting in a duplicate version of MAVA being stored as a dependancy.
```
Comment (DriesSmit) [inline docs/gotchas.md:46]: Just change the MAVA to Mava.

**Review by DriesSmit — CHANGES_REQUESTED** [review]
Thanks @OmaymaMahjoub :slightly_smiling_face: This is a nice addition to the docs.

**Review by AsadJeewa — APPROVED** [review]

**Review by DriesSmit — APPROVED** [review]
Thanks, @OmaymaMahjoub :slightly_smiling_face:

**Pull request merged**